lib/middleware/proxy.js

```diff
@@ -42,11 +43,13 @@ function parseProxyConfig (proxies, config) {
       port = config.port
     }
     const changeOrigin = proxyConfiguration.changeOrigin || false
+    const keepAliveAgent = new http.Agent({ keepAlive: true })
     const proxy = httpProxy.createProxyServer({
       target: { host: hostname, port, https, protocol },
       xfwd: true,
       changeOrigin: changeOrigin,
-      secure: config.proxyValidateSSL
+      secure: config.proxyValidateSSL,
+      agent: keepAliveAgent
```

Comment:
Does this work if target URL is HTTPS?

Reply:
Good catch. I completely missed that. Should be solved now.

Reply:
Another thing is that we probably also need to call `.destroy()` on the agent when proxy server is shut down to make sure sockets are not leaking.

(Although currently Karma is not doing a good job at cleaning up resources after itself it would be good to not add more such places.)

PS I'll take a more thorough look in the coming days as I'm not very familiar with this part of the codebase.

Reply:
Thanks! I'll have a look into ensuring a proper cleanup

Reply:
I added destroying the agents on exit, but need to check how this can be properly unit tested.

Reply:
Thanks! IMO We can ignore this test for now.